server/src/services/semanticHighlight/highlighters/ContractDefinitionHighlighter.ts

```diff
@@ -0,0 +1,17 @@
+import { SemanticTokenTypes } from "vscode-languageserver-protocol";
+import { NodeType, RuleKind, TokenKind } from "@nomicfoundation/slang";
+import { HighlightVisitor } from "../HighlightVisitor";
+import { SlangNode } from "../slangHelpers";
+
+// Highlights contract definitions
+export class ContractDefinitionHighlighter extends HighlightVisitor {
+  public visit(node: SlangNode, _ancestors: SlangNode[]): void {
+    if (
+      node.type === NodeType.Token &&
+      node.kind === TokenKind.Identifier &&
+      _ancestors[_ancestors.length - 1]?.kind === RuleKind.ContractDefinition
```

Comment:
Just to check I am following, this is roughly saying: if I am the identifier that is the immediate child of a contract definition, add me as a ContractDefinition highlight.

Reply:
Exactly. I did it this way instead of using the ContractDefinition's node range because that one includes leading whitespaces and linebreaks, which sometimes break the highlighting (my guess is that vscode doesn't expect a token spanning multiple lines).